### PR TITLE
suppress unittest ResourceWarning messages

### PR DESCRIPTION
**Pull request by dwalton76**
For issue #433 

This does not fix the issue of the filehandles being left open it just suppresses the warnings generated by unittest.  We've had 433 open for almost a year and haven't come up with a good solution and leaving the filehandles open isn't really causing any issues (none that I know of at least).  From googling these unittest ResourceWarning messages we would not be the first repo to just ignore these via a warnings filter.

The ``setUp()`` method just makes it so you can tell what output is coming from what test (this is optional)

new output
```
Import warning: Failed to import "evdev". Button will be unusable!


test_device
===========
	0	outA	./tests/fake-sys/arena/tacho-motor/motor0
	0	in1	./tests/fake-sys/arena/lego-sensor/sensor0
.

test_infrared_sensor
====================
	0	in1	./tests/fake-sys/arena/lego-sensor/sensor0
.

test_joystick_units
===================
	0	outA	./tests/fake-sys/arena/tacho-motor/motor0
	1	outB	./tests/fake-sys/arena/tacho-motor/motor1
.

test_medium_motor
=================
	0	outA	./tests/fake-sys/arena/tacho-motor/motor0
.

test_medium_motor_write
=======================
	0	outA	./tests/fake-sys/arena/tacho-motor/motor0
.

test_motor_on_for_degrees
=========================
	0	outA	./tests/fake-sys/arena/tacho-motor/motor0
.

test_motor_on_for_rotations
===========================
	0	outA	./tests/fake-sys/arena/tacho-motor/motor0
.

test_move_tank_relative_distance
================================
	0	outA	./tests/fake-sys/arena/tacho-motor/motor0
	1	outB	./tests/fake-sys/arena/tacho-motor/motor1
.

test_steering_large_value
=========================
	0	outA	./tests/fake-sys/arena/tacho-motor/motor0
	1	outB	./tests/fake-sys/arena/tacho-motor/motor1
.

test_steering_units
===================
	0	outA	./tests/fake-sys/arena/tacho-motor/motor0
	1	outB	./tests/fake-sys/arena/tacho-motor/motor1
.

test_tank_units
===============
	0	outA	./tests/fake-sys/arena/tacho-motor/motor0
	1	outB	./tests/fake-sys/arena/tacho-motor/motor1
.

test_units
==========
	0	outA	./tests/fake-sys/arena/tacho-motor/motor0
	1	outB	./tests/fake-sys/arena/tacho-motor/motor1
.
----------------------------------------------------------------------
Ran 12 tests in 0.082s

OK
```

### CODE REVIEW

File: tests/api_tests.py [hunk 1]
```diff
@@ -43,6 +43,14 @@ def dummy_wait(self, cond, timeout=None):
 Motor.wait = dummy_wait
 
 class TestAPI(unittest.TestCase):
+
+    def setUp(self):
+        # micropython does not have _testMethodName
+        try:
+            print("\n\n{}\n{}".format(self._testMethodName, "=" * len(self._testMethodName,)))
+        except AttributeError:
+            pass
```
Comment (dwalton76) [inline tests/api_tests.py:66]: This isn't needed to suppress the warnings, I just added it so I could tell what tests were generating the warnings.  I kinda like the output but I can chop this if others do not.

**Comment by dwalton76** [comment]
Also fixes #554

File: tests/README.md [hunk 2]
```diff
@@ -8,20 +8,13 @@ already cloned the ev3dev-lang-python repo you can use the `--recursive` option
...
 ```
 
 # Running Tests
 To run the tests do:
 ```
-$ ./api_tests.py
-```
-
-# Misc
-Commands used to copy the /sys/class node:
```
Comment (WasabiFan) [inline tests/README.md:21]: Why is this removed? Don't we want these instructions still?

File: tests/README.md [hunk 3]
```diff
@@ -8,20 +8,13 @@ already cloned the ev3dev-lang-python repo you can use the `--recursive` option
...
-
-```
-$ node=lego-sensor/sensor0
-$ mkdir -p ./${node}
-$ cp -P --copy-contents -r /sys/class/${node}/* ./${node}/
+$ cd ev3dev-lang-python/
+$ chmod -R g+rw ./tests/fake-sys/devices/**/*
```
Comment (WasabiFan) [inline tests/README.md:18]: Is this necessary? On Travis it was but I still don't fully understand why; it isn't in my local dev environments.

File: .travis.yml [hunk 4]
```diff
@@ -17,7 +17,8 @@ install:
 - if [ $USE_MICROPYTHON = true ]; then ./.travis/install-micropython.sh && PYTHON=~/micropython/ports/unix/micropython; else PYTHON=python3; fi
 script:
 - chmod -R g+rw ./tests/fake-sys/devices/**/*
-- $PYTHON tests/api_tests.py
+- if [ $USE_MICROPYTHON = false ]; then $PYTHON -W ignore tests/api_tests.py; fi
```
Comment (WasabiFan) [inline .travis.yml:20]: Is there a more specific warning filter we can include here? It looks like you can filter on the start of the message and omit all `ResourceWarning`s in particular.

**Comment by WasabiFan** [comment]
Any idea where the `.` comes from after each test?

File: tests/README.md (hunk 2 above)
Comment (dwalton76) [reply tests/README.md:21]: I do not think these are needed anymore....I did not do these steps and am able to run all of the tests successfully.

File: .travis.yml (hunk 4 above)
Comment (dwalton76) [reply .travis.yml:20]: For normal python there is a ``warnings`` module you can import and then filter based on part of the message (this is what I did initially).  micropython does not support it though so the ``-W ignore`` option was the cleanest solution I could come up with.

File: tests/README.md (hunk 3 above)
Comment (dwalton76) [reply tests/README.md:18]: It is, without the ``chmod`` several tests will fail due to
```
ERROR: test_steering_units (__main__.TestAPI)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./tests/api_tests.py", line 300, in test_steering_units
    drive.on_for_rotations(25, SpeedDPS(400), 10)
  File "./tests/../ev3dev2/motor.py", line 1864, in on_for_rotations
    MoveTank.on_for_rotations(self, SpeedNativeUnits(left_speed), SpeedNativeUnits(right_speed), rotations, brake, block)
  File "./tests/../ev3dev2/motor.py", line 1749, in on_for_rotations
    MoveTank.on_for_degrees(self, left_speed, right_speed, rotations * 360, brake, block)
  File "./tests/../ev3dev2/motor.py", line 1779, in on_for_degrees
    self.left_motor._set_rel_position_degrees_and_speed_sp(left_degrees, left_speed_native_units)
  File "./tests/../ev3dev2/motor.py", line 877, in _set_rel_position_degrees_and_speed_sp
    self.position_sp = position_delta
  File "./tests/../ev3dev2/motor.py", line 537, in position_sp
    self._position_sp = self.set_attr_int(self._position_sp, 'position_sp', value)
  File "./tests/../ev3dev2/__init__.py", line 281, in set_attr_int
    return self._set_attribute(attribute, name, str(int(value)))
  File "./tests/api_tests.py", line 31, in _set_attribute
    attribute = _internal_set_attribute(self, attribute, name, value)
  File "./tests/../ev3dev2/__init__.py", line 251, in _set_attribute
    self._raise_friendly_access_error(ex, name)
  File "./tests/../ev3dev2/__init__.py", line 274, in _raise_friendly_access_error
    raise driver_error
  File "./tests/../ev3dev2/__init__.py", line 248, in _set_attribute
    attribute.write(value)
io.UnsupportedOperation: File not open for writing
```

File: tests/README.md (hunk 2 above)
Comment (WasabiFan) [reply tests/README.md:21]: These are steps for adding a new device to the fake-sys system, not for running tests.

Comment (dwalton76) [reply tests/README.md:21]: Should we put these steps in the fake-sys README instead?

Comment (WasabiFan) [reply tests/README.md:21]: I think it might already be there. If it is we don't need it here.

**Comment by dwalton76** [comment]
> Any idea where the `.` comes from after each test?

It is printed by unittest

File: tests/README.md (hunk 2 above)
Comment (dwalton76) [reply tests/README.md:21]: yep it is already in the fake-sys README
